packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -172,74 +39,48 @@ function getEventMap(elm: EventTarget): ListenerMap {
     return listenerInfo;
 }
 
-const shadowRootEventListenerMap: WeakMap<EventListener, WrappedListener> = new WeakMap();
+const shadowRootEventListenerMap: WeakMap<
+    EventListenerOrEventListenerObject,
+    WrappedListener
+> = new WeakMap();
+// Using a WeakMap instead of a WeakSet because this one works in IE11 :(
+const eventsComingFromShadowRoot: WeakMap<Event, 1> = new WeakMap();
+
+function isEventComingFromShadowRoot(event: Event): boolean {
+    return eventsComingFromShadowRoot.has(event);
+}
+
+export function setEventFromShadowRoot(event: Event) {
+    eventsComingFromShadowRoot.set(event, 1);
+}
 
 function getWrappedShadowRootListener(
     sr: SyntheticShadowRootInterface,
-    listener: EventListener
+    listener: EventListenerOrEventListenerObject
 ): WrappedListener {
-    if (!isFunction(listener)) {
-        throw new TypeError(); // avoiding problems with non-valid listeners
-    }
     let shadowRootWrappedListener = shadowRootEventListenerMap.get(listener);
     if (isUndefined(shadowRootWrappedListener)) {
         shadowRootWrappedListener = function(event: Event) {
-            // * if the event is dispatched directly on the host, it is not observable from root
-            // * if the event is dispatched in an element that does not belongs to the shadow and it is not composed,
-            //   it is not observable from the root
-            const { composed } = event as any;
             const target = eventTargetGetter.call(event);
             const currentTarget = eventCurrentTargetGetter.call(event);
-            if (target !== currentTarget) {
-                const rootNode = getRootNodeHost(
-                    target as Node /* because wrapping on shadowRoot */,
-                    {
-                        composed,
-                    }
-                );
-
-                if (
-                    isChildNode(rootNode as HTMLElement, currentTarget as Node) ||
-                    (composed === false && rootNode === currentTarget)
-                ) {
+            // An event that is dispatched directly into the shadowRoot instance vs an event that is dispatched
+            // on the host instance directly are difficult to distinguish. We mark the event on the shadowRoot's
+            // dispatchEvent method, so we know it is a valid event for the shadowRoot.
+            if (target !== currentTarget || isEventComingFromShadowRoot(event)) {
```

Comment:
Looks like we are no longer filtering out events that are not composed with this change...